testdata/mock_server/server.go

```diff
@@ -0,0 +1,92 @@
+package mockServer
+
+import (
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/jarcoal/httpmock"
+)
+
+const (
+	GET  = "GET"
+	POST = "POST"
+)
+
+func CreateAndActivateRestMockServer() {
+	httpmock.Activate()
+
+	httpmock.RegisterResponder(
+		POST, "http://localhost/v1/chain/get_account",
+		httpmock.NewStringResponder(http.StatusOK, OpenFile("chain_get_account.json")),
+	)
+
+	httpmock.RegisterResponder(
+		POST, "http://localhost/v1/chain/get_block",
+		httpmock.NewStringResponder(http.StatusOK, OpenFile("chain_get_block.json")),
+	)
+
+	httpmock.RegisterResponder(
+		POST, "http://localhost/v1/chain/get_block_info",
+		httpmock.NewStringResponder(http.StatusOK, OpenFile("chain_get_block_info.json")),
+	)
+
+	httpmock.RegisterResponder(
+		POST, "http://localhost/v1/chain/get_info",
+		httpmock.NewStringResponder(http.StatusOK, OpenFile("chain_get_info.json")),
+	)
+
+	httpmock.RegisterResponder(
+		POST, "http://localhost/v1/chain/get_block_header_state",
+		httpmock.NewStringResponder(http.StatusOK, OpenFile("chain_get_block_header_state.json")),
+	)
+
+	httpmock.RegisterResponder(
+		POST, "http://localhost/v1/chain/get_abi",
+		httpmock.NewStringResponder(http.StatusOK, OpenFile("chain_get_abi.json")),
+	)
+
+	httpmock.RegisterResponder(
+		POST, "http://localhost/v1/chain/get_currency_balance",
+		httpmock.NewStringResponder(http.StatusOK, OpenFile("chain_get_currency_balance.json")),
+	)
+
+	httpmock.RegisterResponder(
+		POST, "http://localhost/v1/chain/get_currency_stats",
+		httpmock.NewStringResponder(http.StatusOK, OpenFile("chain_get_currency_stats.json")),
+	)
+
+	httpmock.RegisterResponder(
+		POST, "http://localhost/v1/chain/get_required_keys",
+		httpmock.NewStringResponder(http.StatusOK, OpenFile("chain_get_required_keys.json")),
+	)
+
+	httpmock.RegisterResponder(
+		POST, "http://localhost/v1/chain/get_producers",
+		httpmock.NewStringResponder(http.StatusOK, OpenFile("chain_get_producers.json")),
+	)
+
+	httpmock.RegisterResponder(
+		POST, "http://localhost/v1/chain/get_raw_code_and_abi",
+		httpmock.NewStringResponder(http.StatusOK, OpenFile("chain_get_raw_code_and_abi.json")),
+	)
+
+	httpmock.RegisterResponder(
+		POST, "http://localhost/v1/chain/get_table_by_scope",
+		httpmock.NewStringResponder(http.StatusOK, OpenFile("chain_get_table_by_scope.json")),
+	)
+}
+
+func OpenFile(filename string) string {
+	path := []string{"./testdata/mock_server", filename}
+	body, err := os.ReadFile(strings.Join(path, "/"))
+	if err != nil {
+		panic(err)
+	}
+
+	return string(body)
+}
```

Comment:
The hard-coded path currently works because `OpenFile` is used within a test that is at the root of the project. Using it from another sub-package would fail. This is because Go tests are run in their package's directory.

It would be better to pass the full path to it. But this needs to be passed down from `CreateAndActivateRestMockServer`.

To reduce the having to pass `testdata/mock_server` everywhere, I suggest `CreateAndActivateRestMockServer` receives where is located `rootPackageDirectory`.
```suggestion
func OpenFile(rootPackageDirectory string, filename string) string {
	body, err := os.ReadFile(filepath.Join(rootPackageDirectory, "testdata/mock_server", filename))
	if err != nil {
		panic(err)
	}

	return string(body)
}
```

And `CreateAndActivateRestMockServer` would be call like `CreateAndActivateRestMockServer(".")` and `mockserver.OpenFile(".", <filename>)`.

Also, do not use `strings.Join` to join path segment, use `filepath.Join` instead which deals with platform differences.

Reply:
I always do in Python but no knowledge in Go
Thx very much :)

Reply:
https://github.com/eoscanada/eos-go/pull/191/commits/af34b235a84921ad8c77541d4edc9671062da6b6